wallet/src/main/java/com/android/identity_credential/wallet/MainActivity.kt

```diff
@@ -114,12 +136,52 @@ class MainActivity : ComponentActivity() {
         Manifest.permission.NFC to "NFC permission is required to operate"
     ))
 
+    private val permissionsLauncher =
+        registerForActivityResult(ActivityResultContracts.RequestMultiplePermissions()) { permissions ->
+            permissions.entries.forEach {
+                Logger.d(TAG, "permissionsLauncher ${it.key} = ${it.value}")
+                if (!it.value) {
+                    Toast.makeText(
+                        this,
+                        "The ${it.key} permission is required for BLE",
+                        Toast.LENGTH_LONG
+                    ).show()
+                    return@registerForActivityResult
+                }
+            }
+        }
+
+    private val appPermissions: Array<String> =
+        if (android.os.Build.VERSION.SDK_INT >= 31) {
+            arrayOf(
+                Manifest.permission.BLUETOOTH_ADVERTISE,
+                Manifest.permission.BLUETOOTH_SCAN,
+                Manifest.permission.BLUETOOTH_CONNECT,
+            )
+        } else {
+            arrayOf(
+                Manifest.permission.ACCESS_FINE_LOCATION,
+            )
+        }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         application = getApplication() as WalletApplication
 
         permissionTracker.updatePermissions()
+        val permissionsNeeded = appPermissions.filter { permission ->
+            ContextCompat.checkSelfPermission(
+                applicationContext,
+                permission
+            ) != PackageManager.PERMISSION_GRANTED
+        }
+
+        if (permissionsNeeded.isNotEmpty()) {
+            permissionsLauncher.launch(
+                permissionsNeeded.toTypedArray()
+            )
+        }
```

Comment:
This should use`PermissionTracker` which I mentioned in [this comment](https://github.com/openwallet-foundation-labs/identity-credential/pull/469#discussion_r1475969222).

Ideally we'd request the permissions to time-of-use but because we'll need them at NFC tap time we need to ask for the BLE permissions the first time the application is launched (need to be a different `PermissionTracker` instance, I think).